webapp/src/server_fetch_utils.js

```diff
@@ -462,6 +474,20 @@ export function saveUser(user_id, user) {
     });
 }
 
+export function saveRole(user_id, role) {
```

Comment:
We are seeing TypeError/network failures here, I think because `role` is not encapsulated in a JSON object. Perhaps we should just pass `{"role": role}` to the API and go from there